File: Iterator/problem.py
class BrowseHistory:
    __urls = []

    def getUrls(self) -> list:
        return self.__urls

    def push(self, url: str) -> None:
        self.__urls.append(url)

    def pop(self) -> str:
        return self.__urls.pop()


def main():
    history = BrowseHistory()
    history.push("a")
    history.push("b")
    history.push("c")

    for url in history.getUrls():
        print(url)


main()

File: Iterator/problem.dart
class BrowseHistory {
  List<String> _urls = [];

  List<String> get urls => _urls;

  void push(String url) {
    urls.add(url);
  }

  String pop() {
    return urls.removeLast();
  }
}

void main(List<String> args) {
  var history = new BrowseHistory();
  history.push("a");
  history.push("b");
  history.push("c");

  for (var url in history.urls) {
    print(url);
  }
}
